pkg/apis/operator/v1alpha1/common.go

```diff
@@ -33,6 +35,27 @@ const (
 	DeploymentsAvailable apis.ConditionType = "DeploymentsAvailable"
 )
 
+// KComponent is a common interface for accessing meta, spec and status of all known types.
+type KComponent interface {
+	metav1.Object
+	schema.ObjectKind
+
+	// GetSpec returns the common spec for all known types.
+	GetSpec() KComponentSpec
+	// GetStatus returns the common status of all known types.
+	GetStatus() KComponentStatus
+}
+
+// KComponentSpec is a common interface for accessing the common spec of all known types.
+type KComponentSpec interface {
+	// GetConfig returns means to override entries in upstream configmaps.
+	GetConfig() map[string]map[string]string
```

Comment:
nit: i think i'd like to typedef this to `ConfigMapData` just to clarify its purpose and reduce the noise.